pkg/cloud/services/networking/network.go

```diff
@@ -147,6 +150,33 @@ func (s *Service) ReconcileNetwork(openStackCluster *infrav1.OpenStackCluster, c
 	return nil
 }
 
+func (s *Service) DeletePorts(openStackCluster *infrav1.OpenStackCluster) error {
+	networkID := openStackCluster.Spec.Network.ID
+	portList, err := s.client.ListPort(ports.ListOpts{
+		NetworkID:   networkID,
+		DeviceOwner: "",
+	})
+	if err != nil {
+		if capoerrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("searching for no owner port error: %v", err)
+	}
+
+	for _, port := range portList {
+		if strings.Contains(port.Name, openStackCluster.Name) && port.DeviceOwner == "" {
```

Comment:
As the portList should only include ports without a device owner, omitting the condition here should be okay.
```suggestion
		if strings.Contains(port.Name, openStackCluster.Name) {
```